Cargo.toml

```diff
@@ -1,26 +1,22 @@
 [package]
-name = "alacritty"
+name = "alacritty-core"
 version = "0.3.0"
 authors = ["Joe Wilm <joe@jwilm.com>"]
 license = "Apache-2.0"
 build = "build.rs"
-description = "GPU-accelerated terminal emulator"
+description = "GPU-accelerated terminal emulator - library for embedders"
```

Comment:
This is a tiny nitpick, but if the plan is to separate the frontend from the backend completely, there's no more GPU acceleration in alacritty-core and advertising it as such would likely be confusing.

I'd recommend just switching it to something like `Library for writing terminal emulators` (suggestions welcome!).

Reply:
In my view of things, the core would in fact have the GPU acceleration code. Embedders would activate an OpenGL context and then call the core library to render a frame.

Reply:
> In my view of things, the core would in fact have the GPU acceleration code

That seems like an odd choice. I feel like one of the reasons for separating things is so we can have frontends that do not use Alacritty's OpenGL rendering code. This would allow using Alacritty for various things other than writing your own terminal emulator too (like writing a multiplexer).

Reply:
I'm interested in using alacritty as the renderer for my terminal-like iOS app, so I would want to include the GPU code. Other use cases include terminal apps that, say, have a native settings UI. You're right about there being use cases with no graphics, though. I'll go back to the suggestion of an `alacritty_ui` crate for the graphics stuff, so you can use the grid-related code separately.

Reply:
So maybe an `alacritty_ui` crate, an `alacritty_terminal` crate and the `alacritty` crate which is just superglue to hook them both up to each other.

If we're talking priority I'd definitely rank the terminal usecase over the UI one though, since Alacritty's UI is far less complex than what the terminal does. The UI is also very poorly suited for anything other than a terminal emulator due to the limitations that have been imposed to improve performance.

So my main concern is that it wouldn't be possible to have good abstractions for both without any performance hit (which is an absolute requirement), however if both can be pulled out I'd be fine with that too.

Note that my branch already separates an `alacritty` and `alacritty_terminal` subcrate, so I think that would be a decent starting point and first incremental step for this.

Reply:
Agreed, I was going to suggest keeping UI in the alacritty crate for the time being, since that's harder to separate. The terminal part is already fairly cleanly separated from things that depend on UI, I think.

What exactly should the scope be for alacritty_terminal, though? I'll propose something in the main comment thread, since this is getting pretty long for a code comment.

Reply:
> I'll propose something in the main comment thread, since this is getting pretty long for a code conversation.

That's what I was about to recommend too. Keeping things in the main comment thread will likely make it a lot easier to follow the discussion, thanks.